test/extensions/internal_redirect/redirect_extension_integration_test.cc

```diff
@@ -16,6 +16,17 @@ constexpr char kTestHeaderKey[] = "test-header";
 
 class RedirectExtensionIntegrationTest : public HttpProtocolIntegrationTest {
 public:
+  void TearDown() override {
+    for (auto& fake_upstream_connection : upstream_connections_) {
+      AssertionResult result = fake_upstream_connection->close();
+      RELEASE_ASSERT(result, result.message());
+      result = fake_upstream_connection->waitForDisconnect();
+      RELEASE_ASSERT(result, result.message());
+      fake_upstream_connection.reset();
+    }
+    cleanupUpstreamAndDownstream();
+  }
```

Comment:
Out of curiosity, why are these required? Does this current PR tickle this behavior somehow?